client/python/lib/ovmsclient/tfs_compat/http/requests.py

```diff
@@ -0,0 +1,148 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from enum import Enum
+from ovmsclient.tfs_compat.requests import PredictRequest, ModelMetadataRequest, ModelStatusRequest
+
+class HttpPredictRequest(PredictRequest):
+    class Format(Enum):
+        COLUMN = 1
+        COLUMN_NONAME = 2
+        ROW = 3
+        ROW_NONAME = 4
+
+class HttpModelMetadataRequest(ModelMetadataRequest):
+    pass
+
+class HttpModelStatusRequest(ModelStatusRequest):
+    pass
+
+def make_predict_request(inputs, model_name, model_version=0, request_format=HttpPredictRequest.Format.COLUMN):
+    '''
+    Create HttpPredictRequest object.
+
+    Args:
+
+        inputs: Python dictionary in format:
+
+            .. code-block::
+
+                {
+                    ...
+                    <input_name>: <input_data>,
+                    ...
+                }
+            
+            Following types are accepted:
+
+            ============  ==================
+            input_name    | string  
+            input_data    | python scalar,
+                          | python list,
+                          | numpy scalar,
+                          | numpy array
+            ============  ==================
+
+
+        model_name: Name of the model that will receive the request.
+
+        model_version (optional): Version of the model that will receive the request.
+            By default this value is set to 0, meaning the request will be sent to the default version of the model.
+
+        request_format (optional): JSON body structure for the request inputs. Can be one of:
+
+            * *HttpPredictRequest.Format.COLUMN*
+            * *HttpPredictRequest.Format.COLUMN_NONAME*
+            * *HttpPredictRequest.Format.ROW*
+            * *HttpPredictRequest.Format.ROW_NONAME*
+
+            By default this value is set to *HttpPredictRequest.Format.COLUMN*.
+
+    Returns:
+        HttpPredictRequest object filled with **inputs** and target model spec.
+
+    Raises:
+        TypeError:  if unsupported types are provided.
+        ValueError: if arguments have inappropriate values.
+
+    Examples:
+        Request to the default version of the model called "model" that has 2 inputs:  
+
+        >>> predict_request = make_predict_request(
+        ...     inputs={
+        ...         "string_input": "hello there",
+        ...         "numeric_input: np.array([[1, 2, 3], [4, 5, 6]], np.int32)
+        ...     }, 
+        ...     model_name="model")
```

Comment:
format will be set automatically?
how binary data is to be passed? base64 should be handled manually or the same format of arguments like with grpc would be used? For simplicity it might be easier to select the binary data based on string content but for TFS it could be in some model with string input incorrect.

Reply:
Well, in REST we don't really have the possibility to set data type. The data will be set as it's provided in the function argument. Base64 encoded data shall be provided as a string.